spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -154,12 +156,16 @@ public void nack() {
 		catch (RuntimeException re) {
 			if (this.ackMode == AckMode.AUTO) {
 				message.nack();
+				messageNacked = true;
```

Comment:
So, we would only get in here if the sending to the error channel fails? In that case we want to **nack**? Wouldn't that cause repeated re-delivery to no end?

Reply:
I think so. If you have a constant re-delivery, then it is indeed something is wrong in logic of your application.
Fix it - and you are good with acking on error!